### PR TITLE
Add pulsar flink sink connector

### PR DESCRIPTION
**Pull request by aahmed-se**
@XiaoZYang introduces a module for a pulsar sink connector for flink. This PR is moving the work from flink repo to pulsar repo, so the flink connector can be released faster along with Pulsar releases.

Original Flink Github Issue: apache/flink#5845
Jira Issue: https://issues.apache.org/jira/browse/FLINK-9168

Original Author: @XiaoZYang (Zong Yang Xiao)

### CODE REVIEW

File: pulsar-flink/pom.xml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    <groupId>org.apache.pulsar</groupId>
+    <artifactId>pulsar</artifactId>
+    <version>2.2.0-incubating-SNAPSHOT</version>
+    <relativePath>..</relativePath>
+  </parent>
+
+  <artifactId>pulsar-flink</artifactId>
+  <name>Pulsar Flink Connectors</name>
+  <properties>
+    <flink.version>1.6.0</flink.version>
```
Comment (sijie) [inline pulsar-flink/pom.xml]: can you move these properties to root pom file (where is the the central place for placing properties)?

File: pulsar-flink/pom.xml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+              <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+              <artifactSet>
+                <includes>
+                  <include>com.google.guava:guava</include>
+                </includes>
+              </artifactSet>
+              <relocations>
+                <relocation>
+                  <pattern>com.google</pattern>
+                  <shadedPattern>pulsar-storm-shade.com.google</shadedPattern>
```
Comment (sijie) [inline pulsar-flink/pom.xml]: "pulsar-storm-shade" => "pulsar-flink-shade"

**Comment by sijie** [comment]
Created a master ticket for all the efforts related to flink connectors.

**Comment by sijie** [comment]
@aahmed-se : since we are moving other people's PR into pulsar's repo, it is important to mention the original work and credits to the original author @XiaoZYang .

**Comment by sijie** [comment]
retest this please

**Comment by aahmed-se** [comment]
should be fixed

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
run java8 tests

**Comment by chrismiller** [comment]
Looks like PulsarProduceMode.AT_LEAST_ONE is still incorrect (see https://github.com/apache/flink/pull/5845/files/fff36440197e3f34caf3af0b7ed7ba5320003e3b#r187084563)

**Pull request merged**